### PR TITLE
fix: ensure header is not nil

### LINKED ISSUE #39: BUG: Occurs panic
**Describe this problem**
Refer to https://github.com/CeresDB/ceresdb/actions/runs/4435020380/jobs/7781648435?pr=732
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x81dbcd]

goroutine 1 [running]:
github.com/CeresDB/ceresdb-client-go/ceresdb.(*rpcClient).Write(0xc000216600?, {0x9e51e8, 0xc0000300e8}, {0x9[29](https://github.com/CeresDB/ceresdb/actions/runs/4435020380/jobs/7781648435?pr=732#step:8:30)8bd?, 0x7993e5?}, {{0x9249[30](https://github.com/CeresDB/ceresdb/actions/runs/4435020380/jobs/7781648435?pr=732#step:8:31)?, 0x1?}}, {0xc0001926e0, 0x2, 0x2})
	/home/runner/go/pkg/mod/github.com/!ceres!d!b/ceresdb-client-go@v1.1.0/ceresdb/rpc.go:96 +0x12d
github.com/CeresDB/ceresdb-client-go/ceresdb.(*clientImpl).Write(0xc00000e6c0, {0x9e51e8, 0xc0000300e8}, {{{0x924930, 0x6}}, {0xc00007cc80, 0x2, 0x2}})
	/home/runner/go/pkg/mod/github.com/!ceres!d!b/ceresdb-client-go@v1.1.0/ceresdb/client_impl.go:75 +0x49f
main.write({0x9e51e8, 0xc0000300e8}, {0x9e[31](https://github.com/CeresDB/ceresdb/actions/runs/4435020380/jobs/7781648435?pr=732#step:8:32)d8, 0xc00000e6c0}, 0x186e99319b3)
	/home/runner/work/ceresdb/ceresdb/integration_tests/sdk/go/main.go:[36](https://github.com/CeresDB/ceresdb/actions/runs/4435020380/jobs/7781648435?pr=732#step:8:37) +0x46d
main.main()
	/home/runner/work/ceresdb/ceresdb/integration_tests/sdk/go/main.go:118 +0x1a5
exit status 2
```

**Steps to reproduce**

**Expected behavior**

**Additional Information**

### PR DESCRIPTION
**Pull request by chunshao90**
# Which issue does this PR close?

Closes #39 

# Rationale for this change
 Fix bug refer to #39
 

# What changes are included in this PR?
* Fix npe.
* Go mod tidy.

# Are there any user-facing changes?
No.

# How does this change test

### CODE REVIEW

File: ceresdb/rpc.go [hunk 1]
```diff
@@ -50,7 +50,8 @@ func (c *rpcClient) SQLQuery(ctx context.Context, endpoint string, req SQLQueryR
 	if err != nil {
 		return SQLQueryResponse{}, err
 	}
-	if queryResponse.Header.Code != codeSuccess {
+
+	if queryResponse.Header != nil && queryResponse.Header.Code != codeSuccess {
```
Comment (jiacai2050) [inline ceresdb/rpc.go]: when Header is nil, it should return 500 error directly.

File: go.sum [hunk 2]
```diff
@@ -4,10 +4,6 @@ dmitri.shuralyov.com/gpu/mtl v0.0.0-20190408044501-666a987793e9/go.mod h1:H6x//7
 gioui.org v0.0.0-20210308172011-57750fc8a0a6/go.mod h1:RSH6KIUZ0p2xy5zHDxgAM4zumjgTw83q2ge/PI+yyw8=
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
 github.com/BurntSushi/xgb v0.0.0-20160522181843-27f122750802/go.mod h1:IVnqGOEym/WlBOVXweHU+Q+/VP0lqqI8lqeDx9IjBqo=
-github.com/CeresDB/ceresdbproto/golang v0.0.0-20230209095436-43939799b2e6 h1:EqRcuCuMDc+x5vWbyut236dR7RRwWjFOpYjwbi8mNFc=
```
Comment (jiacai2050) [inline go.sum:7]: why this is removed?

Comment (chunshao90) [reply go.sum:7]: Run `go mod tidy`.

File: ceresdb/rpc.go [hunk 3]
```diff
@@ -51,7 +51,14 @@ func (c *rpcClient) SQLQuery(ctx context.Context, endpoint string, req SQLQueryR
 		return SQLQueryResponse{}, err
 	}
 
-	if queryResponse.Header != nil && queryResponse.Header.Code != codeSuccess {
+	if queryResponse.Header != nil {
```
Comment (jiacai2050) [inline ceresdb/rpc.go]: `header == nil`

**Review by jiacai2050 — APPROVED** [review]
lgtm

**Pull request merged**